fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/AdminCheckTabletsCommand.java

```diff
@@ -0,0 +1,105 @@
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.util.PropertyAnalyzer;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.StmtExecutor;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * drop user command
+ */
+public class AdminCheckTabletsCommand extends Command implements Redirect {
+    private final List<Long> tabletIds;
+    private final Map<String, String> properties;
+
+    /**
+     * Check Type in Properties.
+     * At present only CONSISTENCY is supported
+     */
+    public enum CheckType {
+        CONSISTENCY; // check the consistency of replicas of tablet
+        public static CheckType getTypeFromString(String str) throws AnalysisException {
+            try {
+                return CheckType.valueOf(str.toUpperCase());
+            } catch (Exception e) {
+                throw new AnalysisException("Unknown check type: " + str);
+            }
+        }
+    }
+
+    /**
+     * constructor
+     */
+    // ADMIN CHECK TABLET (id1, id2, ...) PROPERTIES ("type" = "check_consistency");
+    public AdminCheckTabletsCommand(List<Long> tabletIds, Map<String, String> properties) {
+        super(PlanType.ADMIN_CHECK_TABLETS_COMMAND);
+        this.tabletIds = tabletIds;
+        this.properties = properties;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        // check auth
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+        }
+
+        if (CollectionUtils.isEmpty(tabletIds)) {
+            throw new AnalysisException("Tablet id list is empty");
+        }
+
+        String typeStr = PropertyAnalyzer.analyzeType(properties);
+        if (typeStr == null) {
+            throw new AnalysisException("Should specify 'type' property");
+        }
+        CheckType checkType = CheckType.getTypeFromString(typeStr);
+
+        if (properties != null && !properties.isEmpty()) {
+            throw new AnalysisException("Unknown properties: " + properties.keySet());
+        }
+
+        if (Objects.requireNonNull(checkType) == CheckType.CONSISTENCY) {
+            Env.getCurrentEnv().getConsistencyChecker().addTabletsToCheck(tabletIds);
+        }
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitAdminCheckTabletsCommand(this, context);
+    }
+
+    @Override
+    public RedirectStatus toRedirectStatus() {
```

Comment:
```suggestion
```
no need to override toRedirectStatus if implements ForwardNoSync

Reply:
Resolved. Please review